### ISSUE TITLE
Standardize notation in URL schemes

### ISSUE THREAD

**Issue by kolgza** [LemmyNet/lemmy#987]
### Proposal

- Change the URL scheme for communities from `https://example.tld/c/community` to `https://example.tld/!community`.
- Change the URL scheme for users from `https://example.tld/u/user` to `https://example.tld/@user`. 

### Reasoning

[From the Lemmy Documentation](https://dev.lemmy.ml/docs/about_guide.html):

> Start typing...
>
> - `@a_user_name` to get a list of usernames.
> - `!a_community` to get a list of communities.

This section of the documentation suggests that the proper syntax to refer to users and communities is `@user` and `!community` respectively. Due to the fact that Lemmy uses Reddit-style URL schemes, I often see people instead use Reddit-style syntax.

### Additional Considerations

This could just be a non-issue.

### Additional context

Mastodon uses the same URL schemes proposed here. For example: `https://mastodon.social/@Gargron/`.

**Comment by dessalines** [comment]
I really dislike this, because have a url prefix with a `/`/ separator to denote the type, is much better than forcing a weird character analyzer. Plus `!` and `#` are protected url characters.

Also we have other types besides those two, such as `/post/`, `/comment/`, etc.

**Comment by kolgza** [comment]
![Screenshot_2020-07-17  c biology — a community about Biology - undefined](https://user-images.githubusercontent.com/46303203/87831106-8f93fa00-c850-11ea-958e-0e3545920fe4.png)
The fact that users often use `/c/community` instead of `!community` suggests Lemmy does not do a good enough job at communicating that `!community` is supposed to be the correct syntax to use when linking to other communities.

**Comment by dessalines** [comment]
I agree that its kinda confusing... pleroma for instance has : https://outerheaven.club/users/vanillabean , which doesn't have an `@` in the url either, yet its pretty consistently `@user@instance` in the UI, just like mastodon, and like we're doing with lemmy.

We want to keep the standard of the rest of the fediverse in mentions and tagging other users, and using `!community@instance` as the community character, even though there won't be community "tagging" IE notifying a whole community, just the autocomplete shortcut. We used to use `/c/` and `/u/` on the front end, but we changed it to be more similar to the rest of the fediverse.
